### ISSUE TITLE
Make `width`, `height` and `alt` required props

### ISSUE THREAD

**Issue by Baroshem** [nuxt/image#1612]
Hey there!

In Nuxt Cloudinary module we have these props as required to promote good practices for images.

If you omit these props, popular tools such as Lighthouse can report this as an issue.

Do you think that we could add these props to Nuxt Image as well as required? Or maybe add a warning in the console that you should add them to have better score and overall :) 

I can create a draft PR with this change if you would like to test how it works :)

**Comment by w33bvGL** [comment]
The console warning is a particularly useful addition. Waiting for the draft

**Comment by Baroshem** [comment]
@danielroe what do you think about that? :)

**Comment by madc** [comment]
There are cases where leaving the alt attribute empty makes sense. For example, decorative images that don't add meaningful context can be hidden from screen readers by using alt="". This helps keep the experience clear and focused for users relying on assistive technologies.

**Comment by Baroshem** [comment]
Thanks for the comment @madc 

What about then checking if an alt is empty but provided in this case? So if a user will not provide prop it will display warning but but if they pass an empty string it will not display a warning? :)

**Comment by razbakov** [comment]
Width and height can be auto generated

**Comment by Baroshem** [comment]
Sounds reasonable.

I wonder if this approach will not indicate to users that they could be using sizes for optimized images (in Lighthouse for example) and the users will try to use while it will no longer be needed.

WDYT?

**Comment by danielroe** [comment]
@razbakov The width/height size doesn't matter (as the actual size of the image can be set in CSS), it's only the aspect ratio that is significant to the browser. Your example assumes we know it's square. If that's the case, you can just set `width` and `height`; this will work fine on any screen size.

More generally, it would be a very nice enhancement to support detecting sizes of images at build time, perhaps only with `ipx` provider.

I would also be up for (as an easy first step) showing a warning in development when height/width are omitted. (Later we can move this to the `nuxt/hints` module - cc: @Baroshem!)

**Comment by chrsigg** [comment]
> More generally, it would be a very nice enhancement to support detecting sizes of images at build time, perhaps only with ipx provider.

Yes please! For `ipx`, `height` and `width` are known during the `build` and could be rendered as attributes to `<img>`, reducing work for the user and also the risk for unintended inconsistencies between the specified and actual image dimensions.

Making `height` and `width` mandatory would prevent that enhancement.

**Comment by MickL** [comment]
I never use `width` and `height`. I always uses `sizes` instead. Possibly also all of them could be omitted to display the original image.

Therefor I dont think any of these values should be required.

**Comment by razbakov** [comment]
> I never use `width` and `height`. I always uses `sizes` instead. Possibly also all of them could be omitted to display the original image.
> 
> Therefor I dont think any of these values should be required.

It's important for SEO to improve Google Pagespeed and to avoid layout shift and for this setting width and height is required

**Comment by MickL** [comment]
How would you set width and height for multiple different screensizes e.g. mobile, tablet, etc?
